### ISSUE TITLE
Does preprocess.py shuffle the training data?

### ISSUE THREAD

**Issue by shawnkx** [pytorch/fairseq#1213]
I am just wondering whether preprocess.py will shuffle the training data?

**Comment by ngoyal2707** [comment]
No, it doesn't.
Shuffling happens during training
